palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/model/AuditablePalisadeRequest.java

```diff
@@ -0,0 +1,133 @@
+
+    @JsonCreator
+    private AuditablePalisadeRequest(
+            final @JsonProperty("palisadeRequest") PalisadeRequest palisadeRequest,
+            final @JsonProperty("auditErrorMessage") AuditErrorMessage auditErrorMessage) {
+        this.palisadeRequest = palisadeRequest;
+        this.auditErrorMessage = auditErrorMessage;
+    }
+
+    /**
+     * Chain any errors from previous stream elements
+     *
+     * @param audit the previous audit or null
+     * @return a new instance of this object
+     */
+    public AuditablePalisadeRequest chain(final AuditErrorMessage audit) {
+        return Optional.ofNullable(audit).map(message -> Builder.create()
+                .withAuditErrorMessage(message))
+                .orElse(this);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final AuditablePalisadeRequest that = (AuditablePalisadeRequest) o;
+        return Objects.equals(palisadeRequest, that.palisadeRequest) &&
+                Objects.equals(auditErrorMessage, that.auditErrorMessage);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(palisadeRequest, auditErrorMessage);
+    }
+
+    public PalisadeRequest getPalisadeRequest() {
+        return palisadeRequest;
+    }
+
+    public AuditErrorMessage getAuditErrorMessage() {
+        return auditErrorMessage;
+    }
+
+    /**
+     * The static builder
+     */
+    public static class Builder {
+
+        /**
+         * The creator function
+         *
+         * @return the composed immutable object
+         */
+        public static IPalisadeRequest create() {
+            return AuditablePalisadeRequest::new;
+        }
+
+        /**
+         * Compose with {@code UserResponse}
+         */
+        public interface IPalisadeRequest {
+            /**
+             * Compose value
+             *
+             * @param audit value or null
+             * @return value object
+             */
+            default AuditablePalisadeRequest withAuditErrorMessage(AuditErrorMessage audit) {
+                return withResponseAndError(null, audit);
+            }
+
+
+            /**
+             * Compose value
+             *
+             * @param request value or null
+             * @return value object
+             */
+            default AuditablePalisadeRequest withPalisadeRequest(PalisadeRequest request) {
+                return withResponseAndError(request, null);
+            }
+
+            /**
+             * Compose value
+             *
+             * @param request          value or null
+             * @param auditErrorMessage value or null
+             * @return value object
+             */
+            AuditablePalisadeRequest withResponseAndError(PalisadeRequest request, AuditErrorMessage auditErrorMessage);
```

Comment:
Looked at your RunnableGraph, so not likely an issue, but worth asking, are you ever going to get an instance with both?